### DISCUSSION TITLE
issues with my esp32

### DISCUSSION THREAD

**Discussion by Hansi137** [jczic/ESP32-MPY-Jama#25] [Q&A] [answered]
Hi, 
i tried out your cool tool but i got a lot of issues when i try to connect to my esp32. I got a connection to the device after i erased my flash with esptool.py via my powershell console and flashed a micropython firmware to it, but after a hard reset of the board i can not connect again via your tool again, but i have no idea what could cause this issue. Maybe you can help.
![image](https://user-images.githubusercontent.com/16804329/221028512-996ea151-8070-4620-b224-f02e08227e62.png)

**Comment by jczic** [comment, marked as answer]
@Hansi137, I may have fixed the problem but for the moment it is not in a binary release but only in the latest git sources.
Check again the issue : https://github.com/jczic/ESP32-MPY-Jama/issues/11 if you want.
Thank you :)

(commit https://github.com/jczic/ESP32-MPY-Jama/commit/d46b4aaf0bc5649f0b8608f741c7e367c661d3de)

**Comment by jczic** [comment]
@Hansi137, I just corrected the problem normally!
If you want, you can try again and confirm me?
However, it's not yet in the MacOS and Windows binaries, you have to clone the repository and run it from the sources.

**Reply by Hansi137** [reply]
Hi,
thanks for the fast reply. Seems to be working now :) What was the issue? I am new and at the beginning to learn programming so i just try out to learn what challanges i am have to struggle with. 
Finaly i have to say that i am at the moment not able to use your tool with all features because i did not understand all the functions at the moment. I had issues with editing and uploading files...For me upycraft worked best to flash firmware and also upload and edit code on my esp32. But i will test with your tool again. I also tried VisualStudio Code with Pymakr extension but also noticed that it is not very user friendy so i did at least all my stuff with upycraft.

**Reply by jczic** [reply]
@Hansi137, thank you for your feedback :)
The connection problem was complex, on some board models and it is now effectively solved.
You can easily edit the files and program your own in the IDE part but you can also configure the Wi-Fi very easily and test things via the Jama Funcs (which are small, ready-made configurable programs).
